src/main/java/org/opensearch/knn/plugin/script/KNNScoringSpace.java

```diff
@@ -144,7 +144,12 @@ public CosineSimilarity(Object query, MappedFieldType fieldType) {
         protected BiFunction<float[], float[], Float> getScoringMethod(final float[] processedQuery) {
             SpaceType.COSINESIMIL.validateVector(processedQuery);
             float qVectorSquaredMagnitude = getVectorMagnitudeSquared(processedQuery);
-            return (float[] q, float[] v) -> 1 + KNNScoringUtil.cosinesimilOptimized(q, v, qVectorSquaredMagnitude);
+            // To be consistent, we will be using same formula used by lucene as mentioned below
+            // https://github.com/apache/lucene/blob/0494c824e0ac8049b757582f60d085932a890800/lucene/core/src/java/org/apache/lucene/index/VectorSimilarityFunction.java#L73
+            return (float[] q, float[] v) -> Math.max(
+                (1.0F + KNNScoringUtil.cosinesimilOptimized(q, v, qVectorSquaredMagnitude)) / 2.0F,
+                0.0F
+            );
```

Comment:
will this create any BWC challenges, mainly in cases when there is a mix cluster. 

Also another thing:
shouldn't this code be 

```suggestion
            // To be consistent, we will be using same formula used by lucene as mentioned below
            // https://github.com/apache/lucene/blob/0494c824e0ac8049b757582f60d085932a890800/lucene/core/src/java/org/apache/lucene/index/VectorSimilarityFunction.java#L73
            return (float[] q, float[] v) -> Math.max(
                (2.0F - KNNScoringUtil.cosinesimilOptimized(q, v, qVectorSquaredMagnitude)) / 2.0F,
                0.0F
            );
```

isn't `KNNScoringUtil.cosinesimilOptimized(q, v, qVectorSquaredMagnitude)` == `rawScore` ?

Reply:
cosinesimilOptimized  = (dotProduct / (Math.sqrt(normalizedProduct)));

Reply:
> will this create any BWC challenges, mainly in cases when there is a mix cluster.
I don't think it will affect the order of neighbors. I believe we should already have bwc test, let me check and confirm

Reply:
> cosinesimilOptimized = (dotProduct / (Math.sqrt(normalizedProduct)));

Oh wait sorry my bad. This formula is after simplification. Can we please add the docs around how we have simplified it? Just for reference in future.

```math
1 - cos\theta = cosineSimil
```

```math
OS Score = (2 - cosineSimil)/2
```

```math
OSScore = (2 - (1 - cos\theta))/2

```

which simplifies to

```math
OSScore = (1 + cos\theta))/2

```

```math
cos\theta = cosinesimilOptimized
```

Reply:
I see your point since this is not about approx search. Let me add version check.  Good Catch @navneet1v

Reply:
updated